packages/components/upload/src/use-handlers.ts

```diff
@@ -25,14 +26,14 @@ const revokeObjectURL = (file: UploadFile) => {
 
 export const useHandlers = (
   props: UploadProps,
+  emit: SetupContext<UploadEmits>['emit'],
   uploadRef: ShallowRef<UploadContentInstance | undefined>
 ) => {
-  const uploadFiles = useVModel(
-    props as Omit<UploadProps, 'fileList'> & { fileList: UploadFiles },
-    'fileList',
-    undefined,
-    { passive: true }
-  )
+  const uploadFiles = ref<UploadFiles>([])
+
+  const updateEventKeyRaw = `onUpdate:fileList` as const
+
+  const hasUpdateHandler = computed(() => isFunction(props[updateEventKeyRaw]))
```

Comment:
Maybe we can change the judgment as follows, then we can just declare `update:fileList` in emits.

```ts
const vnodeProps = getCurrentInstance()!.vnode.props || {}
const hasUpdateHandler = 'onUpdate:fileList' in vnodeProps
```

Reply:
I'm also not sure which is more reasonable. 😅
cc @element-plus/backers

Reply:
I canceled this part.